packages/integrations/vercel/src/serverless/adapter.ts

```diff
@@ -224,6 +259,19 @@ export default function vercelServerless({
 					);
 				}
 			},
+			'astro:server:setup' ({ server }) {
+				// isr functions do not have access to search params, this middleware removes them for the dev mode
+				if (isr) {
+					const exclude_ = typeof isr === "object" ? isr.exclude ?? [] : [];
+					const exclude = exclude_.concat("/_image").map(ex => new RegExp(ex));
```

Comment:
+1 from me too. Also, if a user passes a string, we should do an equality check, and not create a regex.

Reply:
I agree we should be stricter.

It's a RegExp here because that's how vercel processes the routes file.

I'll look into making the exclude paths behave as literal paths.